runtime/service.go

```diff
@@ -145,6 +148,10 @@ type service struct {
 	machineConfig    *firecracker.Config
 	vsockIOPortCount uint32
 	vsockPortMu      sync.Mutex
+
+	// fifos have stdio FIFOs containerd passed to the shim. The key is [taskID][execID].
+	fifos   map[string]map[string]cio.Config
```

Comment:
Since the service struct is in the shim binary and only handling one container, I'm unsure we have a lock contention here.

https://golang.org/pkg/sync/#Map

>  The Map type is specialized. Most code should use a plain Go map instead, with separate locking or coordination, for better type safety and to make it easier to maintain other invariants along with the map content. 

I'd like to keep this as is. We can benchmark later to see whether there is a bottleneck or not.